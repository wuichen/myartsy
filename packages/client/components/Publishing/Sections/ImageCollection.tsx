import { ImagesData } from "Components/Publishing/Typings"
import { find } from "lodash"
import React from "react"
import sizeMe from "react-sizeme"
import styled from "styled-components"
import fillwidthDimensions from "../../../Utils/fillwidth"
import { pMedia } from "../../Helpers"
import { SIZE_ME_REFRESH_RATE } from "../Constants"
import { ArticleLayout, SectionLayout } from "../Typings"
import { Artwork } from "./Artwork"
import { Image } from "./Image"

export interface ImageCollectionProps {
  color?: string
  images: ImagesData
  targetHeight?: number
  gutter?: number
  sectionLayout?: SectionLayout
  articleLayout?: ArticleLayout
  size?: {
    width: number
  }
  fullscreenImages: ImagesData
}

class ImageCollectionComponent extends React.PureComponent<
  ImageCollectionProps,
  null
> {
  static defaultProps = {
    size: {
      width: 680,
    },
  }

  renderImages(dimensions) {
    const {
      articleLayout,
      color,
      gutter,
      images,
      sectionLayout,
      size: { width },
      fullscreenImages,
    } = this.props

    const renderedImages = images.map((image, i) => {
      const url = image.url || image.image

      let imageSize
      if (width <= 600 || dimensions.length === 1) {
        imageSize = {}
      } else {
        imageSize = find(dimensions, ["__id", url])
      }

      const slideshowIndex =
        fullscreenImages &&
        fullscreenImages.findIndex(img => {
          return img.url === image.url
        })

      let renderedImage
      if (image.type === "image") {
        renderedImage = (
          <Image
            image={image}
            color={color}
            sectionLayout={sectionLayout}
            layout={articleLayout}
            width={imageSize.width}
            height={imageSize.height}
            slideshowIndex={slideshowIndex}
          />
        )
      } else if (image.type === "artwork") {
        renderedImage = (
          <Artwork
            artwork={image}
            color={color}
            sectionLayout={sectionLayout}
            layout={articleLayout}
            width={imageSize.width}
            height={imageSize.height}
            slideshowIndex={slideshowIndex}
          />
        )
      } else {
        return false
      }

      const margin = i === dimensions.length - 1 ? 0 : gutter

      return (
        <ImageCollectionItem key={i} margin={margin} width={imageSize.width}>
          {renderedImage}
        </ImageCollectionItem>
      )
    })
    return renderedImages
  }

  render() {
    const { gutter, images, size, targetHeight } = this.props
    const dimensions = fillwidthDimensions(
      images,
      size.width,
      gutter,
      targetHeight
    )
    const renderedImages = this.renderImages(dimensions)

    return <ImageCollectionContainer>{renderedImages}</ImageCollectionContainer>
  }
}

const ImageCollectionContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;

  ${pMedia.xs`
    flex-direction: column;
  `};
`

export const ImageCollectionItem = styled.div<{
  margin?: number
  width?: number
}>`
  margin-right: ${props => (props.margin ? props.margin + "px" : "0px")};
  width: ${props => (props.width ? props.width + "px" : "100%")};
  max-width: 100%;

  img {
    max-width: 100%;
    height: auto;
  }

  ${pMedia.xs`
    margin-bottom: 10px;
    width: 100%;

    img {
      width: 100%;
    }
  `};
`

const sizeMeOptions = {
  refreshRate: SIZE_ME_REFRESH_RATE,
  noPlaceholder: true,
}

export const ImageCollection = sizeMe(sizeMeOptions)(ImageCollectionComponent)
