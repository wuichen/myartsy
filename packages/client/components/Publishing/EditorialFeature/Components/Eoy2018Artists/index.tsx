import { Box, color, Flex, media, Sans, Serif, space } from "@artsy/palette"
import { compact, find, map } from "lodash"
import React from "react"
import styled from "styled-components"

import { unica } from "Assets/Fonts"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { ArticleCardContainer } from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCard"
import {
  ArticleCards,
  ArticlesWrapper,
} from "Components/Publishing/RelatedArticles/ArticleCards/ArticleCards"
import {
  FullLabel,
  ImageSetContainer,
  ImageSetPreview,
  ImgContainer,
} from "Components/Publishing/Sections/ImageSetPreview"
import {
  LabelWrapper,
  SlideshowCta,
  SlideshowTitle,
} from "Components/Publishing/Sections/ImageSetPreview/ImageSetLabel"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { Text } from "Components/Publishing/Sections/Text"
import { resize } from "Utils/resizer"
import { Eoy2018ArticleHeader } from "./ArticleHeader"

export class Eoy2018Artists extends React.Component<EditorialFeaturesProps> {
  getHeaderSections = () => {
    const { sections } = this.props.article
    const headers = []

    sections.map((section, index) => {
      let headerSection
      const sectionAfter = sections[index + 1]
      const sectionAfterisImage =
        sectionAfter && sectionAfter.type === "image_collection"

      if (this.sectionIsHeader(section)) {
        headerSection = {
          index,
          section,
        }
        if (sectionAfterisImage) {
          headerSection.imageSection = sectionAfter
        }
        headers.push(headerSection)
      }
    })
    return headers
  }

  sectionIsHeader = section => {
    const isText = section.type === "text"
    const isHeader = isText && section.body.includes("<h1>")
    return isHeader
  }

  sectionArtistHeader = (section, i) => {
    const headerSections = this.getHeaderSections()
    const { imageSection } = find(headerSections, ["section", section])
    const src = imageSection && imageSection.images[0].url
    const caption = imageSection && imageSection.images[0].caption

    return (
      <ArtistHeaderSection
        key={i}
        mb={40}
        flexDirection={["column", "column", "row", "row"]}
      >
        <ArtistHeaderTitle dangerouslySetInnerHTML={{ __html: section.body }} />
        <ArtistHeaderImg
          src={src && resize(src, { width: 700 })}
          alignItems="flex-end"
        >
          {caption && (
            <Sans size="1" color="white" px={10} py={5}>
              <div dangerouslySetInnerHTML={{ __html: caption }} />
            </Sans>
          )}
        </ArtistHeaderImg>
      </ArtistHeaderSection>
    )
  }

  sectionText = (section, i) => {
    const {
      article: { sections, layout },
      showTooltips,
    } = this.props
    const isChapterStart =
      sections[i - 1] && sections[i - 1].type === "image_collection"

    return (
      <Box maxWidth={["100%", "75%", "75%"]} ml="auto" px={[20, 0]} key={i}>
        <TextSection size="5" pb={[40, 60]} isChapterStart={isChapterStart}>
          <Text
            html={section.body}
            layout={layout}
            showTooltips={showTooltips}
          />
        </TextSection>
      </Box>
    )
  }

  sectionImageSet = (section, i) => {
    return (
      <ImageSetWrapper
        key={i}
        mb={60}
        maxWidth={["100%", "100%", "100%", "75%"]}
      >
        <ImageSetPreview section={section}>
          <CaptionWrapper size={["3", "4"]}>
            <ImageSetCaption
              dangerouslySetInnerHTML={{ __html: section.images[0].caption }}
            />
          </CaptionWrapper>
        </ImageSetPreview>
      </ImageSetWrapper>
    )
  }

  getSections = () => {
    const {
      article: { sections },
    } = this.props
    const headerSections = this.getHeaderSections()
    const headerTextIndexes = map(headerSections, "index")
    const headerImages = map(headerSections, "image")

    const renderedSections = sections.map((section, i) => {
      const isHeader = headerTextIndexes.includes(i)
      const isHeaderImage = headerImages.includes(section)

      if (isHeader) {
        return this.sectionArtistHeader(section, i)
      } else if (isHeaderImage) {
        return null
      } else {
        if (section.type === "text" && i !== 0) {
          return this.sectionText(section, i)
        } else if (section.type === "image_set") {
          return this.sectionImageSet(section, i)
        } else {
          return null
        }
      }
    })
    return renderedSections
  }

  render() {
    const { article, isMobile, isTablet, isTest } = this.props
    const introText = this.sectionText(article.sections[0], 0)
    const headerImages = map(
      compact(map(this.getHeaderSections(), "imageSection")),
      "images"
    )

    return (
      <AricleWrapper>
        <Nav canFix color={color("black100")} backgroundColor="white">
          <NavBorder />
        </Nav>

        <Box px={[10, 10, 55]} maxWidth={1600} mx="auto">
          <Eoy2018ArticleHeader
            images={headerImages}
            isMobile={isMobile}
            isTablet={isTablet}
            isTest={isTest}
          />

          <ArticleContent py={40}>
            <IntroSection
              alignItems="flex-start"
              flexDirection={["column", "row"]}
              pl={[0, 20]}
            >
              <Byline article={article} />
              {introText}
            </IntroSection>
            {this.getSections()}
          </ArticleContent>
        </Box>

        {article.relatedArticles && (
          <Box px={[10, 10, 55]} mb={40} mx="auto">
            <RelatedArticleWrapper>
              <ArticleCards relatedArticles={article.relatedArticles} />
            </RelatedArticleWrapper>
          </Box>
        )}
      </AricleWrapper>
    )
  }
}

const BORDER_WIDTH = 6

const AricleWrapper = styled.div`
  ${NavContainer} {
    border-bottom: none;
  }
`

const NavBorder = styled.div`
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  position: absolute;
  top: 100%;
  left: 55px;
  right: 55px;
  max-width: calc(1600px - 110px);
  margin: 0 auto;

  ${media.md`
    left: 10px;
    right: 10px;
  `};
`

const ArticleContent = styled(Box)`
  border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  blockquote {
    ${unica("s34")};
    line-height: 1.3em;
    margin: 0;
    font-weight: inherit;

    ${media.sm`
      ${unica("s25")};
      line-height: 1.3em;
    `};
  }
`

export const ArtistHeaderSection = styled(Flex)`
  height: 60vh;
  min-height: 450px;
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left-width: 0;

  ${media.md`
    height: fit-content;
  `};
`

const ArtistHeaderTitle = styled.div`
  flex: 1;
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  overflow: hidden;
  min-height: min-content;

  h1 {
    ${unica("s65")};
    width: 100%;
    height: 50%;
    min-height: fit-content;
    padding: ${space(2)}px;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    margin: 0;
    font-weight: inherit;

    ${media.md`
      width: 60%;
      float: right;
      height: 100%;
      border-bottom: none;
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};

    ${media.xs`
      ${unica("s45")};
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
      border-left: none;
      width: 100%;
      height: 50%;
      float: none;
    `};
  }

  h2 {
    ${unica("s25")};
    width: 50%;
    height: 50%;
    min-height: fit-content;
    display: inline-flex;
    padding: ${space(2)}px;
    margin: 0;
    font-weight: inherit;

    &:last-child {
      border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
    }

    ${media.md`
      width: 40%;
      float: left;
      &:last-child {
        border-left: none;
        border-top: ${BORDER_WIDTH}px solid ${color("purple100")};
      }
    `};

    ${media.xs`
      ${unica("s19")};
      width: 50%;
      float: none;
      &:last-child {
        border-left: ${BORDER_WIDTH}px solid ${color("purple100")};
        border-top: none;
      }
    `};
  }

  ${media.md`
    height: 65vh;
    border-right: none;
    border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ArtistHeaderImg = styled(Flex)<{ src?: string }>`
  flex: 1;
  background: ${color("purple100")};

  p {
    text-shadow: 0 0 5px black;
    opacity: 0.6;
  }

  ${props =>
    props.src &&
    `
    background: url(${props.src});
    background-size: cover;
    background-position: center;
  `};

  ${media.md`
    min-height: 70vw;
  `};
`

const IntroSection = styled(Flex)`
  ${BylineContainer} {
    flex-direction: column;
    align-items: flex-start;
    ${media.md`
      padding: 0 20px 20px;
    `};
  }
`

const TextSection = styled(Serif)<{ isChapterStart?: boolean }>`
  ${StyledText} {
    blockquote {
      padding: 0;
    }

    p,
    .paragraph {
      font-size: 24px;
      text-indent: 2em;
      padding: 0;

      ${props =>
        props.isChapterStart &&
        `
        &:first-child {
          text-indent: 0;
        }
    `};
    }
  }
`

const CaptionWrapper = styled(Sans)`
  flex: 1;

  ${media.sm`
    min-width: 50%;
  `};
`

const ImageSetCaption = styled.div`
  height: 100%;
  padding: ${space(2)}px;
  background: white;
  color: black;
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${media.sm`
    border-bottom: none;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  `};
`

const ImageSetWrapper = styled(Box)`
  border: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-left: none;

  ${ImageSetContainer} {
    display: flex;
    flex-direction: row-reverse;

    ${media.sm`
      flex-direction: column-reverse;
    `};
  }

  ${FullLabel} {
    position: relative;
    display: flex;
    flex-direction: column-reverse;
    flex: 1;
    left: auto;
    bottom: auto;
    max-width: inherit;
    border-radius: 0;
    box-shadow: none;

    &:hover {
      background-color: ${color("purple100")};
    }
    ${media.sm`
      flex-direction: row-reverse;

      ${LabelWrapper} {
        max-width: 50%;
      }
    `};
  }

  ${SlideshowTitle} {
    display: none;
  }

  ${SlideshowCta} {
    flex-direction: column;

    > div {
      padding: 0;
    }
  }

  ${ImgContainer} {
    flex: 3;
    border-right: ${BORDER_WIDTH}px solid ${color("purple100")};

    img {
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    ${media.sm`
      border-right: none;
      border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};
    `};
  }
`

const RelatedArticleWrapper = styled.div`
  border-right: ${BORDER_WIDTH}px solid ${color("purple100")};
  border-bottom: ${BORDER_WIDTH}px solid ${color("purple100")};

  ${ArticlesWrapper} {
    margin-bottom: 0;
    margin-top: 0;
  }

  ${ArticleCardContainer} {
    border: none;
  }
`
