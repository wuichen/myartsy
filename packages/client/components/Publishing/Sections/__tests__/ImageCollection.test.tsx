// tslint:disable:no-shadowed-variable

import { Images } from "Components/Publishing/Fixtures/Components"
import { WrapperWithFullscreenContext } from "Components/Publishing/Fixtures/Helpers"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { ImageCollection } from "../ImageCollection"

jest.mock("react-lines-ellipsis/lib/html", () => {
  const React = require("react")
  return () => <div />
})

jest.mock("react-dom/server", () => ({
  renderToStaticMarkup: x => x,
}))

const renderSnapshot = _props => {
  return renderer
    .create(WrapperWithFullscreenContext(<ImageCollection {..._props} />))
    .toJSON()
}

const props = {
  targetHeight: 400,
  gutter: 10,
  images: null,
}
it("renders properly", () => {
  props.images = Images
  const image = renderSnapshot(props)
  expect(image).toMatchSnapshot()
})

it("renders a single image properly", () => {
  props.images = [Images[0]]
  const image = renderSnapshot(props)
  expect(image).toMatchSnapshot()
})
