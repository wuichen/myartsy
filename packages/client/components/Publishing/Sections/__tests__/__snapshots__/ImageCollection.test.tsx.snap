// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single image properly 1`] = `
.c10 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #C2C2C2;
}

.c11 {
  margin-right: 30px;
}

.c12 {
  white-space: nowrap;
}

.c9 {
  padding: 0;
  margin-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 a {
  color: #C2C2C2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9 .title,
.c9 .title a {
  font-style: italic;
}

.c8 {
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.6;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c8:hover {
  opacity: 1;
}

.c7 {
  opacity: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c4 {
  position: relative;
}

.c4:hover .c6 {
  opacity: 1;
}

.c4 .BlockImage__container {
  opacity: 0;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
}

.c4 .image-loaded {
  opacity: 1;
}

.c5 {
  display: block;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin-right: 0px;
  width: 100%;
  max-width: 100%;
}

.c1 img {
  max-width: 100%;
  height: auto;
}

@media (max-width:600px) {
  .c9 {
    padding: 0 10px;
  }
}

@media (max-width:720px) {
  .c4 .c6 {
    opacity: 1;
  }
}

@media (max-width:600px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c1 {
    margin-bottom: 10px;
    width: 100%;
  }

  .c1 img {
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <a
        className="c3"
        href="/artwork/fernando-botero-nude-on-the-beach"
      >
        <div
          className="c4"
        >
          <img
            alt="Nude on the Beach"
            className="BlockImage__container c5"
            height="auto"
            src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=1200&quality=80"
            width="100%"
          />
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={[Function]}
            >
              <svg
                className="expand"
                version="1.1"
                viewBox="0 0 27 27"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="none"
                  strokeWidth="1"
                >
                  <g>
                    <circle
                      cx="13.9575195"
                      cy="13.0588379"
                      fill="#000000"
                      fillRule="nonzero"
                      r="13"
                    />
                    <g
                      stroke="#FFFFFF"
                      transform="translate(8.000000, 7.000000)"
                    >
                      <g>
                        <g
                          transform="translate(0.382532, 0.382532)"
                        >
                          <polyline
                            points="4.78165571 0 11.1571966 0 11.1571966 6.37554094"
                          />
                          <polyline
                            points="-1.01881144e-06 4.78165537 6.37553992 4.78165537 6.37553992 11.1571963"
                            transform="translate(3.187769, 7.969426) rotate(-180.000000) translate(-3.187769, -7.969426) "
                          />
                          <path
                            d="M10.360254,0.796942618 L0.796942618,10.360254"
                            strokeLinecap="square"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
      </a>
      <div>
        <div
          className="c9 c10"
          color="#C2C2C2"
          fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
          fontSize="14px"
        >
          <span
            className="c11"
          >
            <span
              className="c12"
            >
              <a
                href="/artist/fernando-botero"
              >
                Fernando Botero
              </a>
            </span>
          </span>
          <div>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": <span>
                    <span
                      className="title"
                    >
                      <a
                        href="/artwork/fernando-botero-nude-on-the-beach"
                      >
                        Nude on the Beach
                      </a>
                    </span>
                    <span>
                      , 
                    </span>
                    <span
                      className="date"
                    >
                      2000
                    </span>
                  </span>,
                }
              }
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": <span>
                    <a
                      href="/gary-nader"
                    >
                      Gary Nader
                    </a>
                    <span>
                      . 
                    </span>
                    <span
                      className="credit"
                    >
                      Courtesy of Gary Nader
                    </span>
                  </span>,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders properly 1`] = `
.c10 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #C2C2C2;
}

.c11 {
  margin-right: 30px;
}

.c12 {
  white-space: nowrap;
}

.c9 {
  padding: 0;
  margin-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 a {
  color: #C2C2C2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9 .title,
.c9 .title a {
  font-style: italic;
}

.c8 {
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 10px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  opacity: 0.6;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c8:hover {
  opacity: 1;
}

.c7 {
  opacity: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c4 {
  position: relative;
}

.c4:hover .c6 {
  opacity: 1;
}

.c4 .BlockImage__container {
  opacity: 0;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
}

.c4 .image-loaded {
  opacity: 1;
}

.c5 {
  display: block;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 10px 0 10px 0;
}

.c15 {
  padding: 0;
  width: 100%;
  word-break: break-word;
}

.c15 > p,
.c15 p,
.c15 .public-DraftEditorPlaceholder-root,
.c15 .public-DraftStyleDefault-block {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  color: #999;
  margin: 0;
  display: inline;
}

.c15 > a,
.c15 a {
  color: #999;
}

.c15 > a:hover,
.c15 a:hover {
  color: black;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin-right: 10px;
  width: 312px;
  max-width: 100%;
}

.c1 img {
  max-width: 100%;
  height: auto;
}

.c13 {
  margin-right: 10px;
  width: 168px;
  max-width: 100%;
}

.c13 img {
  max-width: 100%;
  height: auto;
}

.c16 {
  margin-right: 0px;
  width: 178px;
  max-width: 100%;
}

.c16 img {
  max-width: 100%;
  height: auto;
}

@media (max-width:600px) {
  .c9 {
    padding: 0 10px;
  }
}

@media (max-width:720px) {
  .c4 .c6 {
    opacity: 1;
  }
}

@media (max-width:600px) {
  .c14 {
    padding: 0px 10px;
  }
}

@media (max-width:600px) {
  .c15 {
    padding: 0px;
  }
}

@media (max-width:600px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c1 {
    margin-bottom: 10px;
    width: 100%;
  }

  .c1 img {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c13 {
    margin-bottom: 10px;
    width: 100%;
  }

  .c13 img {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c16 {
    margin-bottom: 10px;
    width: 100%;
  }

  .c16 img {
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    width={312}
  >
    <div
      className="c2"
    >
      <a
        className="c3"
        href="/artwork/fernando-botero-nude-on-the-beach"
      >
        <div
          className="c4"
        >
          <img
            alt="Nude on the Beach"
            className="BlockImage__container c5"
            height={224}
            src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=1200&quality=80"
            width={312}
          />
          <div
            className="c6 c7"
          >
            <div
              className="c8"
              onClick={[Function]}
            >
              <svg
                className="expand"
                version="1.1"
                viewBox="0 0 27 27"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="none"
                  strokeWidth="1"
                >
                  <g>
                    <circle
                      cx="13.9575195"
                      cy="13.0588379"
                      fill="#000000"
                      fillRule="nonzero"
                      r="13"
                    />
                    <g
                      stroke="#FFFFFF"
                      transform="translate(8.000000, 7.000000)"
                    >
                      <g>
                        <g
                          transform="translate(0.382532, 0.382532)"
                        >
                          <polyline
                            points="4.78165571 0 11.1571966 0 11.1571966 6.37554094"
                          />
                          <polyline
                            points="-1.01881144e-06 4.78165537 6.37553992 4.78165537 6.37553992 11.1571963"
                            transform="translate(3.187769, 7.969426) rotate(-180.000000) translate(-3.187769, -7.969426) "
                          />
                          <path
                            d="M10.360254,0.796942618 L0.796942618,10.360254"
                            strokeLinecap="square"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
      </a>
      <div>
        <div
          className="c9 c10"
          color="#C2C2C2"
          fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
          fontSize="14px"
        >
          <span
            className="c11"
          >
            <span
              className="c12"
            >
              <a
                href="/artist/fernando-botero"
              >
                Fernando Botero
              </a>
            </span>
          </span>
          <div>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": <span>
                    <span
                      className="title"
                    >
                      <a
                        href="/artwork/fernando-botero-nude-on-the-beach"
                      >
                        Nude on the Beach
                      </a>
                    </span>
                    <span>
                      , 
                    </span>
                    <span
                      className="date"
                    >
                      2000
                    </span>
                  </span>,
                }
              }
            />
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": <span>
                    <a
                      href="/gary-nader"
                    >
                      Gary Nader
                    </a>
                    <span>
                      . 
                    </span>
                    <span
                      className="credit"
                    >
                      Courtesy of Gary Nader
                    </span>
                  </span>,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c13"
    width={168}
  >
    <div
      className="article-image"
    >
      <div
        className="c4"
      >
        <img
          alt="Photo by Adam Kuehl for Artsy."
          className="BlockImage__container c5"
          height={224}
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2Fco8j2xq40ROMyBrJQm_4eQ%252FDafenOilPaintingVillage_AK03.jpg&width=1200&quality=80"
          width={168}
        />
        <div
          className="c6 c7"
        >
          <div
            className="c8"
            onClick={[Function]}
          >
            <svg
              className="expand"
              version="1.1"
              viewBox="0 0 27 27"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
                stroke="none"
                strokeWidth="1"
              >
                <g>
                  <circle
                    cx="13.9575195"
                    cy="13.0588379"
                    fill="#000000"
                    fillRule="nonzero"
                    r="13"
                  />
                  <g
                    stroke="#FFFFFF"
                    transform="translate(8.000000, 7.000000)"
                  >
                    <g>
                      <g
                        transform="translate(0.382532, 0.382532)"
                      >
                        <polyline
                          points="4.78165571 0 11.1571966 0 11.1571966 6.37554094"
                        />
                        <polyline
                          points="-1.01881144e-06 4.78165537 6.37553992 4.78165537 6.37553992 11.1571963"
                          transform="translate(3.187769, 7.969426) rotate(-180.000000) translate(-3.187769, -7.969426) "
                        />
                        <path
                          d="M10.360254,0.796942618 L0.796942618,10.360254"
                          strokeLinecap="square"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="c14"
      >
        <div
          className="c15"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Photo by Adam Kuehl for Artsy.</p>",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="c16"
    width={178}
  >
    <div
      className="article-image"
    >
      <div
        className="c4"
      >
        <img
          alt="Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum."
          className="BlockImage__container c5"
          height={224}
          src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2FCpHY-DRr7KW0HGXLslCXHw%2Flarger.jpg&width=1200&quality=80"
          width={178}
        />
        <div
          className="c6 c7"
        >
          <div
            className="c8"
            onClick={[Function]}
          >
            <svg
              className="expand"
              version="1.1"
              viewBox="0 0 27 27"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
                stroke="none"
                strokeWidth="1"
              >
                <g>
                  <circle
                    cx="13.9575195"
                    cy="13.0588379"
                    fill="#000000"
                    fillRule="nonzero"
                    r="13"
                  />
                  <g
                    stroke="#FFFFFF"
                    transform="translate(8.000000, 7.000000)"
                  >
                    <g>
                      <g
                        transform="translate(0.382532, 0.382532)"
                      >
                        <polyline
                          points="4.78165571 0 11.1571966 0 11.1571966 6.37554094"
                        />
                        <polyline
                          points="-1.01881144e-06 4.78165537 6.37553992 4.78165537 6.37553992 11.1571963"
                          transform="translate(3.187769, 7.969426) rotate(-180.000000) translate(-3.187769, -7.969426) "
                        />
                        <path
                          d="M10.360254,0.796942618 L0.796942618,10.360254"
                          strokeLinecap="square"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="c14"
      >
        <div
          className="c15"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Photo by <a href='artsy.net'>Adam Kuehl</a> for Artsy. Image courtesy of the Guggenheim Museum.</p>",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
